src/Storefront/Framework/Cache/CacheKeyProviderInterface.php

```diff
@@ -0,0 +1,20 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Storefront\Framework\Cache;
+
+use Symfony\Component\HttpFoundation\Request;
+
+interface CacheKeyProviderInterface
```

Comment:
This should be an abstract class

`AbstractCacheKeyProvider`.

See https://github.com/shopware/platform/blob/trunk/src/Core/Content/Seo/AbstractSeoResolver.php as example